mono/eglib/gqsort.c

```diff
@@ -40,14 +40,14 @@ typedef struct _QSortStack {
 	size_t count;
 } QSortStack;
 
-#define QSORT_PUSH(sp, a, c) (sp->array = a, sp->count = c, sp++)
+#define QSORT_PUSH(sp, a, c) (sp->array = g_cast (a), sp->count = c, sp++)
 #define QSORT_POP(sp, a, c) (sp--, a = sp->array, c = sp->count)
 
 #define SWAPTYPE(TYPE, a, b) {              \
-	long __n = size / sizeof (TYPE);    \
-	register TYPE *__a = (TYPE *) (a);  \
-	register TYPE *__b = (TYPE *) (b);  \
-	register TYPE t;                    \
+	gssize __n = size / sizeof (TYPE);  \
+	TYPE *__a = (TYPE *) (a);           \
```

Comment:
What's wrong with `register`?

Reply:
‘Register’ is a C++ warning at least, maybe error now or in future.

Reply:
`long` there was insidious -- having the same-ish meaning on all but one platform -- win64.